### PR TITLE
Fix pre/post-conditions on RVector.alloc_reserve

### PR DESCRIPTION
**Pull request by wintersteiger**
Used by Merkle trees, see also `LowStar.Vector.alloc_reserve`.

### CODE REVIEW

File: ulib/LowStar.RVector.fst [hunk 1]
```diff
@@ -757,11 +757,14 @@ val alloc_reserve:
...
     (requires (fun h0 -> true))
     (ensures (fun h0 rv h1 ->
-      modifies (V.loc_vector rv) h0 h1 /\
+      modifies loc_none h0 h1 /\
       rv_inv h1 rv /\
       V.frameOf rv = rid /\
       V.size_of rv = 0ul /\
-      S.equal (as_seq h1 rv) S.empty))
+      S.equal (as_seq h1 rv) S.empty /\
+      Set.equal (Map.domain (HS.get_hmap h0))
```
Comment (msprotz) [inline ulib/LowStar.RVector.fst]: this post-condition is unusual, did you mean to use the `Stack` effect?

Comment (wintersteiger) [reply ulib/LowStar.RVector.fst]: I don't understand the difference - I just need it to be the same as on `LowStar.Vector.alloc_reserve`. Would some of this be implicit with the `Stack` effect?

Comment (wintersteiger) [reply ulib/LowStar.RVector.fst]: (Yes; will simplify both.)

Comment (wintersteiger) [reply ulib/LowStar.RVector.fst]: Actually, no, `Stack` requires `same_refs_in_all_regions m0 m1`, which isn't true here. Should it be true or is there a different effect that doesn't require that?

**Comment by aseemr** [comment]
@wintersteiger, @msprotz, can you please update the PR (resolve merging with master, else close if it's not needed?).

**Comment by aseemr** [comment]
Closing this PR since there hasn't been any activity on it, please reopen whenever you get a chance to rework it.

**Pull request closed without merging**